src/doorman/Channel.cpp

```diff
@@ -0,0 +1,393 @@
+    }
+    // Log the source hostname but mask it for privacy.
+    log_out("ID: Looked up %s", get_masked_hostname(hostName).c_str());
+}
+
+static bool SupportsAnsi(const char *src) {
+    static const char *terms[] = {"xterm",      "xterm-colour", "xterm-color", "ansi",
+                                  "xterm-ansi", "vt100",        "vt102",       "vt220"};
+    size_t i;
+    for (i = 0; i < (sizeof(terms) / sizeof(terms[0])); ++i)
+        if (!strcasecmp(src, terms[i]))
+            return true;
+    return false;
+}
+
+void Channel::closeConnection() {
+    // Log the source IP but masked for privacy.
+    log_out("[%d] Closing connection to %s", id_, get_masked_hostname(hostname_).c_str());
+    close();
+}
+
+void Channel::set_echo(bool echo) {
+    SendCom(fd_, echo ? WONT : WILL, TELOPT_ECHO);
+    echoing_ = echo;
+}
+
+void Channel::on_reconnect_attempt() {
+    if (!fd_.is_open())
+        return;
+    if (!sent_reconnect_message_) {
+        send_to_client("Attempting to reconnect to Xania"sv);
+        sent_reconnect_message_ = true;
+    }
+    send_to_client(".", 1);
+}
+
+void Channel::close() {
+    fd_.close();
+    host_lookup_fd_.close();
+    if (host_lookup_pid_) {
+        if (::kill(host_lookup_pid_, 9) < 0)
+            log_out("Couldn't kill child process (ident process has already exited)");
+        host_lookup_pid_ = 0;
+    }
+}
+
+void Channel::newConnection(Fd fd, sockaddr_in address) {
+    // One day I will be a constructor...
+    this->fd_ = std::move(fd);
+    hostname_ = inet_ntoa(address.sin_addr);
+    port_ = ntohs(address.sin_port);
+    netaddr_ = ntohl(address.sin_addr.s_addr);
+    buffer_.clear();
+    width_ = 80;
+    height_ = 24;
+    ansi_ = got_term_ = false;
+    echoing_ = true;
+    sent_reconnect_message_ = false;
+    connected_ = false;
+    auth_char_name_.clear();
+
+    log_out("[%d] Incoming connection from %s on fd %d", id_, get_masked_hostname(hostname_).c_str(), fd_.number());
+
+    // Send all options out
+    sendTelopts();
+
+    // Tell them if the mud is down
+    if (!mud_->connected()) {
+        auto msg = "Xania is down at the moment - you will be connected as soon "
+                   "as it is up again.\n\r"sv;
+        send_to_client(msg.data(), msg.size());
+    }
+
+    sendConnectPacket();
+
+    // Start the async hostname lookup process.
+    int pipe_fds[2];
+    if (pipe(pipe_fds) == -1) {
+        log_out("Pipe failed!");
+        return;
+    }
+    Fd parent_fd(pipe_fds[0]);
+    Fd child_fd(pipe_fds[1]);
+    log_out("Pipes %d<->%d", pipe_fds[0], pipe_fds[1]);
+    auto forkRet = fork();
+    switch (forkRet) {
+    case 0: {
+        // Child process - go and do the lookup
+        log_out("ID: doing lookup");
+        async_lookup_process(address, std::move(child_fd));
+        log_out("ID: exiting");
+        exit(0);
+    } break;
+    case -1:
+        // Error - unable to fork()
+        log_out("Unable to fork() an IdentD lookup process");
```

Comment:
I think  all the identd strings can be taken out (now or later)

Reply:
Agreed; as part of the logging changes I'd like to make I think `log_out` will die and then we'll have `debug`, `info`, `warn` levels etc so it's not quite so noisy when we don't want it to be. I'll have a rethink then.